### PR TITLE
Integrate shared table collapsing js into iOS app

### PR DESCRIPTION
**Pull request by montehurd**
https://phabricator.wikimedia.org/T158686

The first commit allows us to easily pull in common js (and other) bits used by both iOS and Android.
Stephen set up repro for these common bits here: https://github.com/wikimedia/applib

The second commit fixes an error which was happening when we ran Grunt (which processes all of the files in the "www" directory and generates the "assets" directory files which get bundled).

### CODE REVIEW

File: www/package.json [hunk 1]
```diff
@@ -3,6 +3,9 @@
   "description": "JS files required for the Wikipedia app",
   "repository": "https://gerrit.wikimedia.org/r/apps/ios/wikipedia",
   "version": "0.1.0",
+  "dependencies": {
+       "applib": "0.1.2"
```
Comment (niedzielski) [inline www/package.json]: Consider merging [this guy](https://github.com/wikimedia/applib/pull/1) and then you can use the new API

Comment (montehurd) [inline www/package.json]: Done!

**Comment by niedzielski** [comment]
Since adding an NPM depenedency will have no functional change in itself, please consider adding the file changes for the to `collapseTables.js` and the generated Grunt bundle as well

**Comment by montehurd** [comment]
@niedzielski I was hoping to keep follow-on functional changes separate to simplify shepherding this PR through code review.

Oh was [this](https://phabricator.wikimedia.org/T158309#3059605) the ticket pattern you suggested, or does it sound good?

**Comment by niedzielski** [comment]
:+1: @montehurd whatever makes sense as long as it's coming!

**Comment by montehurd** [comment]
@niedzielski 

Ok I updated the title of https://phabricator.wikimedia.org/T158686 to...

"Integrate shared **table collapsing** js into iOS app" 

... and I added a commit to this PR integrating the 1st method "getTableHeader".

File: www/js/transforms/collapseTables.js [hunk 2]
```diff
@@ -1,42 +1,6 @@
...
-        //recurse into children of this element
-        var ret = getTableHeader(el);
-        //did we get a list of TH from this child?
-        if (ret.length > 0) {
-            thArray = thArray.concat(ret);
-        }
-    }
-    return thArray;
-}
+var getTableHeader = require("applib").CollapseElement.getTableHeader;
```
Comment (montehurd) [inline www/js/transforms/collapseTables.js]: `getTableHeader` was the first method we moved to the common [applib](https://github.com/wikimedia/applib) repo 

We then removed `getTableHeader` from our `www/js/transforms/collapseTables.js` iOS source file, then ran `make grunt` to rebuild the bundled `assets` folder contents and confirmed the version we removed disappeared from `Wikipedia/assets/index.js` and the version we added to [applib](https://github.com/wikimedia/applib) appeared in `Wikipedia/assets/index.js`.

File: www/package.json [hunk 3]
```diff
@@ -4,7 +4,7 @@
   "repository": "https://gerrit.wikimedia.org/r/apps/ios/wikipedia",
   "version": "0.1.0",
   "dependencies": {
-       "applib": "0.1.2"
+       "applib": "1.0.0"
```
Comment (montehurd) [inline www/package.json]: I updated `applib` version to `1.0.0`, then ran `make grunt`, which caused `Wikipedia/assets/index.js` to be rebuilt with upstream changes.

File: www/js/transforms/collapseTables.js [hunk 4]
```diff
@@ -1,6 +1,6 @@
 var transformer = require("../transformer");
 var utilities = require("../utilities");
-var getTableHeader = require("applib").CollapseElement.getTableHeader;
+var getTableHeader = require("applib").CollapseTable.getTableHeader;
```
Comment (montehurd) [inline www/js/transforms/collapseTables.js]: Here I updated our source to reference the name change from the last commit. I then ran `make grunt` again to regenerate `Wikipedia/assets/index.js`.

**Comment by montehurd** [comment]
Closing this because I need to update some our "devDependencies" versions. Will do the bits from this PR as follow-ons.

**Pull request closed without merging**